odoo/addons/base/models/ir_http.py

```diff
@@ -464,12 +467,19 @@ def binary_content(self, xmlid=None, model='ir.attachment', id=None, field='raw'
         :param str default_mimetype: default mintype if no mintype found
         :param str access_token: optional token for unauthenticated access
                                  only available  for ir.attachment
+        :param str allow_xaccel: optional argument to bypass xaccel NGINX serving
+                                and fetch raw data instead of only a header
         :returns: (status, headers, content)
         """
         record, status = self._get_record_and_check(xmlid=xmlid, model=model, id=id, field=field, access_token=access_token)
+        xaccel_header = False
 
         if not record:
-            return (status or 404, [], None)
+            return status or 404, [], None
+
+        if odoo.tools.config['xaccel'] and hasattr(record, "store_fname") and record.store_fname and allow_xaccel:
```

Comment:
Why is the flag check the *last* one here despite being a straight and trivial boolean?

Also why is this checked on every access despite only making sense for attachments? There's already a dedicated path for attachments.